sql/src/test/java/io/crate/planner/PlannerTest.java

```diff
@@ -1898,8 +1898,22 @@ public void testLimitThatIsBiggerThanPageSizeCausesQTFPUshPlan() throws Exceptio
     }
 
     @Test
-    public void testKillPlan() throws Exception {
-        Plan killPlan = plan("kill all");
+    public void testKillPlanAll() throws Exception {
+        KillPlan killPlan = (KillPlan) plan("kill all");
         assertThat(killPlan, instanceOf(KillPlan.class));
+        assertThat(killPlan.jobId(), notNullValue());
+        assertThat(killPlan.jobToKill().isPresent(), is(false));
+    }
+
+    @Test
+    public void testKillPlanJobs() throws Exception {
+        KillPlan killJobsPlan = (KillPlan) plan("kill '6a3d6fb6-1401-4333-933d-b38c9322fca7'");
+        assertThat(killJobsPlan.jobId(), notNullValue());
+        assertThat(killJobsPlan.jobToKill().get().toString(), is("6a3d6fb6-1401-4333-933d-b38c9322fca7"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testKillPlanJobsIdIsNotParsable() throws Exception {
+        plan("kill '6a3d6fb6-1401-4333-93'");
```

Comment:
imho this test actually just re-tests the analyzer because it will never hit the planner?